src/xhr.js

```diff
@@ -59,6 +61,35 @@ class Xhr {
       return assertSuccess(response).json();
     });
   }
+
+  postJson(input, init) {
+    const initType = Object.prototype.toString.call(init);
+
+    assert(initType == '[object Object]',
+        'init must be an Object instance or an object literal. Got %s',
+        initType);
+
+    const bodyType = Object.prototype.toString.call(init.body);
+
+    // Assume JSON strict mode where only objects or arrays are allowed
+    // as body.
+    assert(['[object Object]', '[object Array]'].indexOf(bodyType) > -1,
+        'body must be of type object or array. %s',
+        init.body);
+
+    // Clobber method and headers properties so we can guarantee the
+    // operation is a JSON post.
+    init.method = 'POST';
+    init.headers = {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json;charset=UTF-8'
+    };
+    init.body = JSON.stringify(init.body);
+
+    return this.fetch_.call(null, input, init).then(response => {
```

Comment:
ah gotcha, so we want to treat it as a pure function with no this binding just to be safe? can we bind null early on the constructor instead? (just so we can do it one place)